common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java

```diff
@@ -53,7 +58,12 @@ public void onTitleScreenInit(TitleScreenInitEvent.Post e) {
     public void onTitleScreenInit(ScreenInitEvent e) {
         if (!(e.getScreen() instanceof TitleScreen titleScreen)) return;
 
-        addWynncraftButton(titleScreen);
+        if (!initBootPassed && autoConnect.get()) {
+            autoConnect();
+        } else {
+            addWynncraftButton(titleScreen);
+        }
+        initBootPassed = true;
```

Comment:
What is the purpose of this? Doesn't the autoConnect() function already set this to be true?

Reply:
Having inside of the onTitleScreenInit() is crucial - it ensures that if the user goes back to the Title-Screen by any means - they will not get autoConnected to the server.

technically we can take it out in the autoConnect() method.
kristofbolyai said "Just set this before you start connecting so this only happens once?" 
i think may or may not ensure to not try to connect multiple times at once or in rapid succession.

Reply:
I see. Then, we could put it before that if statement and remove the one inside autoConnect()?

Reply:
I think the best thing would be, removing it from autoConnect() and leaving it in the onTitleScreenInit() but after the if statement - if put above. it would never trigger autoConnect().

Another option would be splitting the if statement which consists of the config-check && initBootPassed and move the initBootPassed check to autoConnect()